pkg/services/accesscontrol/evaluator/evaluator.go

```diff
@@ -36,21 +41,30 @@ func evaluateScope(dbScopes map[string]struct{}, targetScopes ...string) (bool,
 		return true, nil
 	}
 
+	logger := log.New("accesscontrol.permissioncheck")
 	for _, s := range targetScopes {
+
 		var match bool
 		for dbScope := range dbScopes {
+			if strings.ContainsAny(s, "*?") {
+				msg := fmt.Sprintf("Invalid scope: %v should not contain meta-characters like * or ?", s)
+				logger.Error(msg)
+				return false, errors.New(msg)
+			}
 			rule, err := glob.Compile(dbScope, ':', '/')
 			if err != nil {
 				return false, err
 			}
 
 			match = rule.Match(s)
 			if match {
+				logger.Debug(fmt.Sprintf("Access control: matched scope %v against permission %v", s, dbScope))
 				return true, nil
 			}
 		}
 	}
 
+	logger.Debug(fmt.Sprintf("Access control: failed!  Could not match scopes  %v against permissions %v", targetScopes, dbScopes))
```

Comment:
When using the logger interface, let the first message be a short identifying message and then add what you want to include for context as variadic key-value parameters to make sure they're consistent with the rest of Grafana:

```suggestion
	logger.Debug("no matching permission found",
	"requiredPermissions", targetScopes,
	"userPermissions", dbScopes,
	)
```